openedx/features/wikimedia_features/meta_translations/api/v0/utils.py

```diff
@@ -0,0 +1,220 @@
+                "data":{
+                    "display_name":"Problem 1",
+                    "content":""
+                }
+            },
+            "B36BXKA90A56Y5QI":{
+                "usage_key":"block-v1:Arbisoft+CS101+TranslatedRerunUrdu+type@html+block@16563a84fc80420cb0fb24d823d5a4f5",
+                "category":"html",
+                "data_block_ids: {
+                    "display_name": 1,
+                    "content": 10
+                },
+                "data": {
+                    "display_name":"Html Text",
+                    "content":"<h1>Hello World<h1/>"
+                }
+            },
+        },
+
+        ABCDASDAGSEEESD0:{
+            "7VLKTTPX1ZUJI8KA":{
+                "usage_key":"block-v1:Arbisoft+CS101+TranslatedRerunUrdu+type@problem+block@fa094274fa444ba2b348a1a00cb117da",
+                "category":"problem",
+                "data": {
+                    "display_name":"SDAFADASD 1",
+                    "content":""
+                }
+            },
+            "B36BXKA90A56Y5QI":{
+                "usage_key":"block-v1:Arbisoft+CS101+TranslatedRerunUrdu+type@html+block@16563a84fc80420cb0fb24d823d5a4f5",
+                "category":"html",
+                "data": {
+                    "display_name":"HERA TESD",
+                    "content":""
+                }
+            },
+        }
+    }
+    """
+    if depth == 0 or not hasattr(block, 'children'):
+        random_key = get_random_string()
+        data, base_data = get_block_data_from_table(block, wiki_objects)
+        data_map = { random_key: data }
+        base_data_map = { random_key: base_data }
+        return data_map, base_data_map
+    
+    random_key = get_random_string()
+    data, base_data = get_block_data_from_table(block, wiki_objects)
+    data_map = { random_key: data }
+    base_data_map = { random_key: base_data }
+
+    data_map[random_key]['children'] = {}
+    base_data_map[random_key]['children'] = {}
+
+    for child in block.get_children():
+        course_outline, course_base_outline = get_recursive_blocks_data_from_table(child, wiki_objects, depth - 1)
+        data_map[random_key]['children'].update(course_outline)
+        base_data_map[random_key]['children'].update(course_base_outline)
+    return data_map, base_data_map
+
+def get_course_outline(course_id, N=3):
+    """
+    Get course outline of a course
+    Arguments:
+        course_id: Course ID
+        N: depth
+    Return:
+        dict: structured outline of a course based on depth
+    """
+    wiki_objects = WikiTranslation.objects.filter(target_block__course_id=course_id)
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_by_id(course_key)
+    course_data, base_course_data = get_recursive_blocks_data_from_table(course, wiki_objects, N)
+    return course_data, base_course_data
+
+def get_outline_course_to_units(course):
+    """
+    Get course and base course outline from course to units
```

Comment:
```suggestion
    Get course and base course outline from course-block to depth: unit blocks.
    It will not include components level data (i.e problems, html etc) in returned outline. 
```